conan/cli/commands/test.py

```diff
@@ -38,30 +38,36 @@ def test(conan_api, parser, *args):
     out.info("Profile build:")
     out.info(profile_build.dumps())
 
+    deps_graph = run_test(conan_api, path, ref, profile_host, profile_build, remotes, lockfile,
+                          args.update, build_modes=None)
+    save_lockfile_out(args, deps_graph, lockfile, cwd)
```

Comment:
This is a small change, not a pure refactor: the lockfile is saved only after the test is successful. If the test fails, no lockfile is created. I think it is better than having a failing test, yet had created a lockfile, wdyt?